windows-agent/internal/proservices/landscape/executor.go

```diff
@@ -254,61 +252,47 @@ func installFromURL(ctx context.Context, homeDir string, distro gowsl.Distro, ro
 	return nil
 }
 
-func download(w io.Writer, url string) (err error) {
-	defer func() {
-		if err != nil {
-			err = fmt.Errorf("could not download %q: %v", url, err)
-		}
-	}()
+func download(ctx context.Context, f io.Writer, url, checksum string) (err error) {
+	defer decorate.OnError(&err, "could not download %q", url)
 
 	//nolint:gosec // G107 expects url to be const, but we are reading it from Landscape.
 	resp, err := http.Get(url)
 	if err != nil {
 		return err
 	}
-	defer resp.Body.Close()
 
 	if resp.StatusCode >= 400 {
 		return fmt.Errorf("http request failed with code %d", resp.StatusCode)
 	}
 
-	if _, err := io.Copy(w, resp.Body); err != nil {
+	// Verify checksum and write file to disk
+	reader := io.TeeReader(resp.Body, f)
```

Comment:
nitpick: this is clearly one case, where a `r` for the reader variable is clear enough and don’t collide with potential generic/package name.